gcc/rust/rust-session-manager.cc

```diff
@@ -783,6 +784,9 @@ Session::injection (AST::Crate &crate)
 void
 Session::expansion (AST::Crate &crate)
 {
+  /* We need to name resolve macros and imports here */
+  Resolver::EarlyNameResolver ().go (crate);
```

Comment:
We might need to add something like this to how we load_extern_crate's but lets not worry about that yet.

We don't do any expansion within load_extern_crate because when we are exporting metadata we should have stripped all nodes as expected and macros all expanded. We do need to export macros so we need some kind of method to allow for referencing these macros at some point but yeah lets not worry about that yet.